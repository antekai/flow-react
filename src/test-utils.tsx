import { RenderHookResult, render, renderHook } from "@testing-library/react";

interface ProvidersForTestsProps {
  children: JSX.Element;
}

export const ProvidersForTests = ({ children }: ProvidersForTestsProps) => {
  return <div>{children}</div>;
};

const customRender = (ui: JSX.Element, options?: Record<string, unknown>) =>
  render(ui, {
    wrapper: ({ children }) => (
      <ProvidersForTests {...options}>{children}</ProvidersForTests>
    ),
  });

export const customRenderHook = <P, R>(
  hook: (props: P) => R,
  options?: Record<string, unknown>
): RenderHookResult<any, any> =>
  renderHook(hook, {
    wrapper: ({ children }) => (
      <ProvidersForTests {...options}>{children}</ProvidersForTests>
    ),
  });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
export { customRenderHook as renderHook };
