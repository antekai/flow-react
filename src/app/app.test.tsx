import { App } from ".";
import { render } from "../test-utils";

describe("App", () => {
  const { asFragment } = render(<App />);

  it("should render successfully", () => {
    expect(asFragment()).toMatchSnapshot();
  });
});
