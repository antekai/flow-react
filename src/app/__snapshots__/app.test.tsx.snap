// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > should render successfully 1`] = `
<DocumentFragment>
  <div
    class="StyledWorkflow-Xslek eWuHGn"
  >
    <div
      class="StyledSelect-cfRtAz jKPwck css-b62m3t-container"
    >
      <span
        class="css-1f43avz-a11yText-A11yText"
        id="react-select-2-live-region"
      />
      <span
        aria-atomic="false"
        aria-live="polite"
        aria-relevant="additions text"
        class="css-1f43avz-a11yText-A11yText"
      />
      <div
        class=" css-13cymwt-control"
      >
        <div
          class=" css-1fdsijx-ValueContainer"
        >
          <div
            class=" css-1jqq78o-placeholder"
            id="react-select-2-placeholder"
          >
            Search and add nodes
          </div>
          <div
            class=" css-qbdosj-Input"
            data-value=""
          >
            <input
              aria-autocomplete="list"
              aria-describedby="react-select-2-placeholder"
              aria-expanded="false"
              aria-haspopup="true"
              autocapitalize="none"
              autocomplete="off"
              autocorrect="off"
              class=""
              id="react-select-2-input"
              role="combobox"
              spellcheck="false"
              style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
              tabindex="0"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class=" css-1hb7zxy-IndicatorsContainer"
        >
          <span
            class=" css-1u9des2-indicatorSeparator"
          />
          <div
            aria-hidden="true"
            class=" css-1xc3v61-indicatorContainer"
          >
            <svg
              aria-hidden="true"
              class="css-tj5bde-Svg"
              focusable="false"
              height="20"
              viewBox="0 0 20 20"
              width="20"
            >
              <path
                d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
              />
            </svg>
          </div>
        </div>
      </div>
      <div>
        <input
          name="search-nodes"
          type="hidden"
          value=""
        />
      </div>
    </div>
    <div
      class="StyledViewFlow-jDENTs bJSSWT"
    >
      <div
        class="react-flow"
        data-testid="rf__wrapper"
        style="width: 100%; height: 100%; overflow: hidden; position: relative; z-index: 0;"
      >
        <div
          class="react-flow__renderer"
          style="position: absolute; width: 100%; height: 100%; top: 0px; left: 0px;"
        >
          <div
            class="react-flow__pane"
            style="position: absolute; width: 100%; height: 100%; top: 0px; left: 0px;"
          >
            <div
              class="react-flow__viewport react-flow__container"
              style="transform: translate(0px,0px) scale(1);"
            >
              <svg
                class="react-flow__edges react-flow__container"
                height="500"
                style="z-index: 0;"
                width="500"
              >
                <defs />
                <g />
              </svg>
              <div
                class="react-flow__edgelabel-renderer"
              />
              <div
                class="react-flow__nodes"
                style="position: absolute; width: 100%; height: 100%; top: 0px; left: 0px;"
              />
            </div>
          </div>
        </div>
        <svg
          class="react-flow__background"
          data-testid="rf__background"
          style="position: absolute; width: 100%; height: 100%; top: 0px; left: 0px;"
        >
          <pattern
            height="20"
            id="pattern-1undefined"
            patternTransform="translate(-0.5,-0.5)"
            patternUnits="userSpaceOnUse"
            width="20"
            x="0"
            y="0"
          >
            <circle
              cx="0.5"
              cy="0.5"
              fill="#91919a"
              r="0.5"
            />
          </pattern>
          <rect
            fill="url(#pattern-1undefined)"
            height="100%"
            width="100%"
            x="0"
            y="0"
          />
        </svg>
        <div
          class="react-flow__panel react-flow__controls bottom left"
          data-testid="rf__controls"
          style="pointer-events: all;"
        >
          <button
            aria-label="zoom in"
            class="react-flow__controls-button react-flow__controls-zoomin"
            title="zoom in"
            type="button"
          >
            <svg
              viewBox="0 0 32 32"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M32 18.133H18.133V32h-4.266V18.133H0v-4.266h13.867V0h4.266v13.867H32z"
              />
            </svg>
          </button>
          <button
            aria-label="zoom out"
            class="react-flow__controls-button react-flow__controls-zoomout"
            title="zoom out"
            type="button"
          >
            <svg
              viewBox="0 0 32 5"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h32v4.2H0z"
              />
            </svg>
          </button>
          <button
            aria-label="fit view"
            class="react-flow__controls-button react-flow__controls-fitview"
            title="fit view"
            type="button"
          >
            <svg
              viewBox="0 0 32 30"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3.692 4.63c0-.53.4-.938.939-.938h5.215V0H4.708C2.13 0 0 2.054 0 4.63v5.216h3.692V4.631zM27.354 0h-5.2v3.692h5.17c.53 0 .984.4.984.939v5.215H32V4.631A4.624 4.624 0 0027.354 0zm.954 24.83c0 .532-.4.94-.939.94h-5.215v3.768h5.215c2.577 0 4.631-2.13 4.631-4.707v-5.139h-3.692v5.139zm-23.677.94c-.531 0-.939-.4-.939-.94v-5.138H0v5.139c0 2.577 2.13 4.707 4.708 4.707h5.138V25.77H4.631z"
              />
            </svg>
          </button>
          <button
            aria-label="toggle interactivity"
            class="react-flow__controls-button react-flow__controls-interactive"
            title="toggle interactivity"
            type="button"
          >
            <svg
              viewBox="0 0 25 32"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M21.333 10.667H19.81V7.619C19.81 3.429 16.38 0 12.19 0c-4.114 1.828-1.37 2.133.305 2.438 1.676.305 4.42 2.59 4.42 5.181v3.048H3.047A3.056 3.056 0 000 13.714v15.238A3.056 3.056 0 003.048 32h18.285a3.056 3.056 0 003.048-3.048V13.714a3.056 3.056 0 00-3.048-3.047zM12.19 24.533a3.056 3.056 0 01-3.047-3.047 3.056 3.056 0 013.047-3.048 3.056 3.056 0 013.048 3.048 3.056 3.056 0 01-3.048 3.047z"
              />
            </svg>
          </button>
        </div>
        <div
          class="react-flow__panel react-flow__attribution bottom right"
          data-message="Please only hide this attribution when you are subscribed to React Flow Pro: https://pro.reactflow.dev"
          style="pointer-events: all;"
        >
          <a
            aria-label="React Flow attribution"
            href="https://reactflow.dev"
            rel="noopener noreferrer"
            target="_blank"
          >
            React Flow
          </a>
        </div>
        <div
          id="react-flow__node-desc-1"
          style="display: none;"
        >
          Press enter or space to select a node.You can then use the arrow keys to move the node around. Press delete to remove it and escape to cancel. 
        </div>
        <div
          id="react-flow__edge-desc-1"
          style="display: none;"
        >
          Press enter or space to select an edge. You can then press delete to remove it or escape to cancel.
        </div>
        <div
          aria-atomic="true"
          aria-live="assertive"
          id="react-flow__aria-live-1"
          style="position: absolute; width: 1px; height: 1px; margin: -1px; border: 0px; padding: 0px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); clip-path: inset(100%);"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
